src/main/java/mate/academy/library/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.library.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.library.dto.category.CategoryDto;
+import mate.academy.library.exception.EntityNotFoundException;
+import mate.academy.library.mapper.CategoryMapper;
+import mate.academy.library.model.Category;
+import mate.academy.library.repository.category.CategoryRepository;
+import mate.academy.library.service.CategoryService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find category by id: " + id)
+        );
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto save(CategoryDto categoryDto) {
+        Category category = categoryMapper.toModel(categoryDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public CategoryDto update(Long id, CategoryDto categoryDto) {
+        Category category = categoryMapper.toModel(categoryDto);

```

Comment:
```
Category category = repository.findById(id)
mapper.updateCategoryFromDto(categoryDto, category);
return mapper.toDto(repository.save(category));
```

Where in CategoryMapper 
`void updateCategoryFromDto(CategoryRequestDto categoryDto, @MappingTarget Category category);`